packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -239,63 +242,32 @@ export const Select = <T,>(props: SelectProps<T>): JSX.Element => {
     style,
     menuClassName,
     menuStyle,
-    onShow,
-    onHide,
-    popoverProps,
     multiple = false,
     triggerProps,
     status,
+    popoverProps,
     ...rest
   } = props;
 
-  const [isOpenState, setIsOpen] = React.useState(false);
-  const isOpen = popoverProps?.visible ?? isOpenState;
+  const [isOpen, setIsOpen] = React.useState(false);
 
-  const [minWidth, setMinWidth] = React.useState(0);
   const [liveRegionSelection, setLiveRegionSelection] = React.useState('');
 
   const selectRef = React.useRef<HTMLDivElement>(null);
 
-  const onShowHandler = React.useCallback(
-    (instance: PopoverInstance) => {
-      setIsOpen(true);
-      onShow?.(instance);
-    },
-    [onShow],
-  );
-
-  const onHideHandler = React.useCallback(
-    (instance: PopoverInstance) => {
-      setIsOpen(false);
-      selectRef.current?.focus({ preventScroll: true }); // move focus back to select button
-      onHide?.(instance);
-    },
-    [onHide],
-  );
-
-  React.useEffect(() => {
-    if (selectRef.current) {
-      setMinWidth(selectRef.current.offsetWidth);
-    }
-  }, [isOpen]);
-
-  const onKeyDown = (event: React.KeyboardEvent) => {
-    if (event.altKey) {
+  const show = React.useCallback(() => {
+    if (disabled) {
       return;
     }
+    setIsOpen(true);
+    popoverProps?.onVisibleChange?.(true);
+  }, [disabled, popoverProps]);
 
-    switch (event.key) {
-      case 'Enter':
-      case ' ':
-      case 'Spacebar': {
-        setIsOpen((o) => !o);
-        event.preventDefault();
-        break;
-      }
-      default:
-        break;
-    }
-  };
+  const hide = React.useCallback(() => {
+    setIsOpen(false);
+    selectRef.current?.focus({ preventScroll: true });
+    popoverProps?.onVisibleChange?.(false);
```

Comment:
nit: The old comment was helpful to understand this code. So better to bring it back.

```suggestion
    selectRef.current?.focus({ preventScroll: true }); // move focus back to select button
```